provider/gateway/rest/server.go

```diff
@@ -67,3 +69,21 @@ func NewJwtServer(ctx context.Context,
 
 	return srv, nil
 }
+
+func NewResourceServer(ctx context.Context,
+	log log.Logger,
+	serverAddr string,
+	providerAddr sdk.Address,
+	cert *x509.Certificate,
+	lokiGwAddr string,
+) (*http.Server, error) {
+	srv := &http.Server{
+		Addr:    serverAddr,
+		Handler: newResourceServerRouter(log, providerAddr, cert.PublicKey.(*ecdsa.PublicKey), lokiGwAddr),
```

Comment:
Why are we casting uncondtionally on this line to a `*ecdsa.PublicKey` ?

Reply:
Fixed